pkg/declextract/interface.go

```diff
@@ -60,7 +64,29 @@ func (ctx *context) processFunctions() {
 			ctx.funcs[fn.Name] = fn
 		}
 	}
+	coverBlocks := make(map[string][]*cover.Block)
+	for _, file := range ctx.coverage {
+		for _, fn := range file.Functions {
+			coverBlocks[file.FilePath+fn.FuncName] = fn.Blocks
+		}
+	}
 	for _, fn := range ctx.Functions {
+		for _, block := range coverBlocks[fn.File+fn.Name] {
+			var match *FunctionScope
+			for _, scope := range fn.Scopes {
+				if scope.Arg == -1 {
+					match = scope
+				}
+				if block.FromLine >= scope.StartLine && block.FromLine <= scope.EndLine {
```

Comment:
`block.ToLine <= scope.EndLine`? Or it doesn't matter?

Reply:
Currently we have Block.FromLine == Block.ToLine. If they are different, then we can have a block that does not fully belong to any scope. I think it's better/simpler to use FromLine and attribute it to one scope deterministically.